tools/genhtml.pl

```diff
@@ -72,16 +73,22 @@ ()
 
 sub get_background_colour_column() {
     if ( $detected_fail == 1 ) {
+      $test_result = "FAIL";
       return "#ff0000";
     } elsif ( $detected_brok == 1 ) {
+      $test_result = "BROK";
       return Yellow;
     } elsif ( $detected_warn == 1 ) {
+      $test_result = "WARN";
       return Fuchsia;
     } elsif ( $detected_retr == 1 ) {
+      $test_result = "RETR";
       return "#8dc997";
     } elsif ( $detected_conf == 1 ) {
+      $test_result = "CONF";
       return Aqua;
     } else {
+      $test_result = "PASS";
       return "#66ff66";
     }
 }
```

Comment:
Passing the value in a global variable is ugly and doing that in a function whose name indicates that it only returns color even more so.